### PR TITLE
Netty cleanup

### PR DESCRIPTION
**Pull request by jyemin**
https://jira.mongodb.org/browse/JAVA-3511

Evergreen patch: https://evergreen.mongodb.com/version/5dcf1d8532f4174b892203b0

### CODE REVIEW

File: .evergreen/run-tests.sh [hunk 1]
```diff
@@ -24,11 +25,8 @@ TOPOLOGY=${TOPOLOGY:-server}
 COMPRESSOR=${COMPRESSOR:-}
 SLOW_TESTS_ONLY=${SLOW_TESTS_ONLY:-false}
```
Comment (rozza) [inline .evergreen/run-tests.sh]: Add: `STREAM_TYPE=${STREAM_TYPE:-nio2}`

File: .evergreen/run-tests.sh [hunk 2]
```diff
@@ -24,11 +25,8 @@ TOPOLOGY=${TOPOLOGY:-server}
...
 
-# JDK6 needs async.type=netty
-if [ "$JDK" == "jdk6" ]; then
-  export ASYNC_TYPE="-Dorg.mongodb.async.type=netty"
-else
-  export ASYNC_TYPE="-Dorg.mongodb.async.type=nio2"
+if [ "$STREAM_TYPE" == "netty" ]; then
```
Comment (rozza) [inline .evergreen/run-tests.sh]: By setting a default above you don't need the if statement: 

`export ASYNC_TYPE="-Dorg.mongodb.test.async.type=$STREAM_TYPE"`

File: driver-core/src/test/functional/com/mongodb/ClusterFixture.java [hunk 3]
```diff
@@ -349,7 +349,7 @@ public static StreamFactory getStreamFactory() {
     }
 
     public static StreamFactory getAsyncStreamFactory() {
-        String streamType = System.getProperty("org.mongodb.async.type", "nio2");
+        String streamType = System.getProperty("org.mongodb.test.async.type", "nio2");
```
Comment (rozza) [inline driver-core/src/test/functional/com/mongodb/ClusterFixture.java]: I think you may need to set the property in the top level:  build.gradle

Comment (jyemin) [reply driver-core/src/test/functional/com/mongodb/ClusterFixture.java]: Yep, I neglected to change the name there.  Fixed in latest commit, and tested manually from command line to make sure the system property was getting set properly.

File: .evergreen/run-tests.sh (hunk 1 above)
Comment (jyemin) [reply .evergreen/run-tests.sh]: Done

File: .evergreen/run-tests.sh (hunk 2 above)
Comment (jyemin) [reply .evergreen/run-tests.sh]: Done

**Comment by jyemin** [comment]
New patch: https://evergreen.mongodb.com/version/5dd40aac0305b9334ec81dfa

**Comment by jyemin** [comment]
New patch: https://evergreen.mongodb.com/version/5dd40e0a0ae60619752ae7f7

**Pull request closed without merging**